docs/source/trainer/using_the_trainer.rst

```diff
@@ -0,0 +1,484 @@
+:class:`.State`). This allows algorithms to modify the state at the
+various events above.
+
+.. seealso::
+
+    :doc:`Events<trainer/events>` and :class:`.State`
+
+Algorithms
+~~~~~~~~~~
+
+The Composer trainer is designed to easily apply our library of
+algorithms to both train more efficiently and build better models. These
+can be enabled by passing the appropriate algorithm class to ``algorithms``
+argument.
+
+.. testcode::
+
+   from composer import Trainer
+   from composer.algorithms import LayerFreezing, Mixup
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='160ep',
+                     algorithms=[
+                         LayerFreezing(freeze_start=0.5, freeze_level=0.1),
+                         Mixup(alpha=0.1),
+                     ])
+
+   # the algorithms will automatically be applied during the appropriate
+   # points of the training loop
+   trainer.fit()
+
+We handle inserting those algorithms into the training loop and in the
+right order.
+
+.. seealso::
+
+    Our :doc:`Algorithms<algorithms>` guide, and the individual
+    :doc:`Method Cards </method_cards/methods_overview>` for each algorithm.
+
+
+Optimizers & Schedulers
+-----------------------
+
+You can easily specify which optimizer and learning rate scheduler to
+use during training. Composer provides a library of various optimizers
+and schedulers, but you can also include one of your own.
+
+.. code:: python
+
+   from composer import Trainer
+   from composer.models import ComposerResNet
+   from composer.optim.scheduler import cosine_annealing_scheduler
+   from torch.optim import SGD
+
+   model = ComposerResNet(model_name="resnet50", num_classes=1000)
+   optimizer = SGD(model.parameters(), lr=0.1)
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='90ep',
+                     optimizer=optimizer,
+                     scheduler=consine_annealing_scheduler)
+
+.. seealso::
+
+    :doc:`Schedulers`<trainer/optimizers_and_schedulers>
+
+Training on GPU
+~~~~~~~~~~~~~~~
+
+Control which device you use for training with the ``device`` parameter,
+and we will handle the data movement and other systems-related
+engineering. We currently support the ``cpu`` and ``gpu`` devices.
+
+.. code:: python
+
+   from composer import Trainer
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='160ep',
+                     device='gpu')
+
+Distributed Training
+~~~~~~~~~~~~~~~~~~~~
+
+It's also simple to do data-parallel training on multiple GPUs. Composer
+provides a launcher command that works with the trainer and handles all
+the ``torch.distributed`` setup for you.
+
```

Comment:
Should we comment about the need for a `DistributedSampler` in the dataloader?

Reply:
In addition, should we mention that, unlike grad accum, the dataloader batch size should not be scaled by the number of devices?